ethcore/src/error.rs

```diff
@@ -167,6 +167,8 @@ pub enum BlockError {
 	UnknownParent(H256),
 	/// Uncle parent given is unknown.
 	UnknownUncleParent(H256),
+	/// The same author issued different votes at the same step.
+	DoubleVote(H160)
```

Comment:
trailing comma missing.

one nice thing in a future redesign of the engine trait might involve an associated `Error` type to convey potential consensus errors (which are unique to each engine) as opposed to bundling all possible consensus errors in with block errors.

@gavofyork this is one of the use-cases of the diverging `!` being promoted to a full type -- you might have something like an `InstantSeal` engine where consensus errors are not possible, and this makes it possible to specify it in the type-system.